### PR TITLE
Fix default Collapse behavior / Add Form constructor

### PR DESCRIPTION
**Pull request by 0x3333**
If you use a Collapse with toggle = true when it is loaded it should not animate the expansion.

In the onLoad() method there is a call to:
```java
collapse(getElement(), toggle);
```
We should change to:
```java
if (this.toggle) {
    addStyleNames("in");
}
```
This way the element will be presented already expanded. In case the user wants it to be animated, he should call ```show()```.

Here is a demo of plain bootstrap behaviour: http://jsfiddle.net/uf61hdv2/1/

### CODE REVIEW

File: gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java [hunk 1]
```diff
@@ -36,6 +37,15 @@
  */
 public class Form extends AbstractForm implements HasType<FormType> {
 
+    public Form() {
```
Comment (jgodi) [inline gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java]: Do we need the empty constructor?

**Comment by jgodi** [comment]
Doesn't ``` <b:PanelCollapse in="true">``` do the same thing?

File: gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java (hunk 1 above)
Comment (ArloL) [inline gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java]: It's the equivalent of calling Form(FormType.DEFAULT).

I like it better like this

~~~
public Form() {
    this(FormType.DEFAULT);
}

@UiConstructor
public Form(final FormType type) {
    setType(type);
}
~~~

Makes it a little more obvious.

File: gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java [hunk 2]
```diff
@@ -36,6 +37,15 @@
...
+    }
+
+    @UiConstructor
+    public Form(final FormType type) {
+      this();
```
Comment (ArloL) [inline gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java]: I think the indentation is wrong here. Seems like you're using 2 spaces.

**Comment by ArloL** [comment]
I am not sure that this should be in one PR. Could you perhaps split it?

Then it's easier to reason about the changes. I know this case is really small but it's a good habit.

**Comment by ArloL** [comment]
> Doesn't `<b:PanelCollapse in="true">` do the same thing?

It might but I think Collapse should behave correctly nonetheless.

File: gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java (hunk 1 above)
Comment (0x3333) [inline gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java]: Ok.

File: gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java (hunk 2 above)
Comment (0x3333) [inline gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java]: My bad.

File: gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java (hunk 1 above)
Comment (0x3333) [inline gwtbootstrap3/src/main/java/org/gwtbootstrap3/client/ui/Form.java]: We need the empty constructor otherwise you must provide the FormType, and this will break the API.

**Comment by 0x3333** [comment]
You are right. I'll split.

**Comment by ArloL** [comment]
Great!

**Comment by yuppyavd** [comment]
Commit https://github.com/gwtbootstrap3/gwtbootstrap3/commit/fc0c5c9040dfcdf1a74c19cea52e355e3531cf42 introduced other unexpected behavior. I have many Collapse elements with toggle="false" by default. When I first change to any view with Collapse element, it is hidden. But when I changed to some other view and go back to previous view, Collapse element is visible. After second change view to another and back again, Collapse is hidden. Between view changes toggle/show/hide methods on Collapse are not called.

I have changed code in Collapse onLoad method and it works correctly:
```java
if (this.toggle) {
    addStyleNames("in");
} else {
    collapse(getElement(), toggle);
}
```

**Comment by 0x3333** [comment]
TL;DR; version: I may be wrong, but I believe that you have some issue in your code.

Long version:
Actually the default behavior is to animate only if changing states(visible/hidden). When you set toggle before onLoad you are setting the default state. Calling collapse will animate the transition. That's why we changed it in the first place.

See the documentation in bootstrap: http://getbootstrap.com/javascript/#collapse-methods. The constructor has a toggle option which will configure the initial state of the component. That's the behavior in the onLoad.

If you need to show/hide after, you call collapse.
